packages/shared_preferences/shared_preferences_android/lib/src/shared_preferences_async_android.dart

```diff
@@ -31,22 +37,34 @@ base class SharedPreferencesAsyncAndroid
   /// Returns a SharedPreferencesPigeonOptions for sending to platform.
   SharedPreferencesPigeonOptions _convertOptionsToPigeonOptions(
       SharedPreferencesOptions options) {
+    if (options is SharedPreferencesAsyncAndroidOptions) {
+      return SharedPreferencesPigeonOptions(
+        fileName: options.originalSharedPreferencesOptions?.fileName,
+        useDataStore:
+            options.backend == SharedPreferencesAndroidBackendLibrary.DataStore,
+      );
+    }
     return SharedPreferencesPigeonOptions();
   }
 
+  SharedPreferencesAsyncApi _getApiForBackend(
+      SharedPreferencesPigeonOptions options) {
+    return options.useDataStore ? _dataStoreApi : _sharedPreferencesApi;
```

Comment:
The use of the same API with different suffixes so you could switch in a single place is very clever!